lib/executor/shared_iterations.go

```diff
@@ -193,12 +193,32 @@ func (si SharedIterations) Run(ctx context.Context, out chan<- stats.SampleConta
 	defer activeVUs.Wait()
 
 	regDurationDone := regDurationCtx.Done()
-	runIteration := getIterationRunner(si.executionState, si.logger, out)
+	runIteration := getIterationRunner(si.executionState, si.logger)
+
+	vusDone := make(chan struct{})
+	activeVUsCount := new(uint64)
+	// Wait for all VUs to finish
+	defer func() {
+		for i := uint64(0); i < atomic.LoadUint64(activeVUsCount); i++ {
+			<-vusDone
+		}
+	}()
```

Comment:
This seems wrong for the same reasons I mentioned in the per-vu-iterations

Reply:
Should be resolved in the latest push.